src/main/java/org/terasology/blockdetector/systems/BlockDetectorSystemImpl.java

```diff
@@ -326,9 +327,9 @@ public void detectBlocks() {
             int minDistance = Integer.MAX_VALUE;
             for (Vector3i block : detectedBlocks) {
                 int distance = (int) Math.sqrt(
-                        Math.pow(block.getX() - playerPosition.getX(), 2)
-                                + Math.pow(block.getY() - playerPosition.getY(), 2)
-                                + Math.pow(block.getZ() - playerPosition.getZ(), 2));
+                        Math.pow(block.x() - playerPosition.x(), 2)
+                                + Math.pow(block.y() - playerPosition.y(), 2)
+                                + Math.pow(block.z() - playerPosition.z(), 2));
```

Comment:
I'd use streams API here and make use for `block.distance(playerPosition)`. We could also take the `sqrt` out of the loop and only do it once in the end.